### PR TITLE
Improve BHoM Analytics Pop Up Window 

### LINKED ISSUE #36: Improve BHoM Analytics Pop-UP Window
#### Description:

- Improve UX by modifying per UI
- Allow movable window
- Additional options for non project work
- Improve UI by including branding

### PR DESCRIPTION
**Pull request by travispotterBH**
New PR for branch name compliance: https://github.com/BHoM/BHoMAnalytics_Toolkit/pull/37

### Issues addressed by this PR

Closes #36

Added Branding:
image
@FraserGreenroyd  I need help knowing where the image should be placed, and referenced. Will need to ship with installer. 
Image below:
BHoM_Logo

Added custom UI for Revit:
BhomAnalytics - Revit version

Revit Self Assign Project Number, when Project Information is set:
BhomAnalytics - Revit Self Assign Project Number

Revit User Assign Project Number, when Project Information is not set:
BhomAnalytics - Revit user Assign Project Number

Added custom UI for Grasshopper:
BhomAnalytics - Grasshopper version

### Changelog

- Improved UX by modifying per UI
- Allows movable window
- Includes additional options for non project work
- Improves UI by including branding

### Additional comments

@FraserGreenroyd @al-fisher requested modifications to the installer to update information about logging. 
image

### CODE REVIEW

File: BHoMAnalytics_UI/CaptureProjectData.xaml.cs [hunk 1]
```diff
@@ -43,58 +43,148 @@ namespace BH.UI.Analytics
...
+            NonProjectListBox.ItemsSource = Enum.GetValues(typeof(NonProjectOption));
+            this._uiName = uiName;
+            
+
+            if (_uiName == "Grasshopper")
+            {
+                UIProtipText.Visibility = Visibility.Visible;
+                UIProtipText.Text = "Protip: Set use in ProjectNumber panel to avoid getting this popup.";
+            }
+            if (_uiName == "Revit")
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_UI/CaptureProjectData.xaml.cs:60]:
```suggestion
            }
            else if (_uiName == "Revit")
```

File: BHoMAnalytics_UI/CaptureProjectData.xaml.cs [hunk 2]
```diff
@@ -43,58 +43,148 @@ namespace BH.UI.Analytics
...
+            {
+                UIProtipText.Visibility = Visibility.Visible;
+                BHoMUseTextBlock.Text = "Please indicate purpose of this use of BHoM Tools for Revit";
+                UIProtipText.Text = "Protip: Set Project Number in Revit settings to avoid getting this popup.";
+            }
+            if (_uiName == "Excel")
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_UI/CaptureProjectData.xaml.cs:66]:
```suggestion
            }
            else if (_uiName == "Excel")
```

File: BHoMAnalytics_UI/CaptureProjectData.xaml.cs [hunk 3]
```diff
@@ -43,58 +43,148 @@ namespace BH.UI.Analytics
...
+        #region Constructor
+        public CaptureProjectData(string uiName)
         {
             InitializeComponent();
-            this.ShowDialog();
+            VersionTextBlock.Text = $"BHoM Version: {BH.Engine.Base.Query.BHoMVersion()}";
+            NonProjectListBox.ItemsSource = Enum.GetValues(typeof(NonProjectOption));
+            this._uiName = uiName;
+            
+
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_UI/CaptureProjectData.xaml.cs:54]:
```suggestion
```

File: BHoMAnalytics_UI/CaptureProjectData.xaml.cs [hunk 4]
```diff
@@ -43,58 +43,148 @@ namespace BH.UI.Analytics
...
     {
-        public CaptureProjectData()
+        #region Constructor
+        public CaptureProjectData(string uiName)
         {
             InitializeComponent();
-            this.ShowDialog();
+            VersionTextBlock.Text = $"BHoM Version: {BH.Engine.Base.Query.BHoMVersion()}";
+            NonProjectListBox.ItemsSource = Enum.GetValues(typeof(NonProjectOption));
+            this._uiName = uiName;
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_UI/CaptureProjectData.xaml.cs:52]:
```suggestion
            _uiName = uiName;
```

File: BHoMAnalytics_UI/CaptureProjectData.xaml.cs [hunk 5]
```diff
@@ -43,58 +43,148 @@ namespace BH.UI.Analytics
...
-            this.Height = 250;
+            this.Height = 380;
             ProjectIDInput.Focus();
         }
-
+        private void Click_ConfirmProjectBtn(object sender, EventArgs e)
+        {
+            ConfirmProject();
+        }
+        private void KeyDown_ProjectIDEntry(object sender, KeyEventArgs e)
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_UI/CaptureProjectData.xaml.cs:96]:
```suggestion

        private void Click_ConfirmProjectBtn(object sender, EventArgs e)
        {
            ConfirmProject();
        }
        
        private void KeyDown_ProjectIDEntry(object sender, KeyEventArgs e)
```

File: BHoMAnalytics_UI/CaptureProjectData.xaml.cs [hunk 6]
```diff
@@ -43,58 +43,148 @@ namespace BH.UI.Analytics
...
+                projectID = $"Non-Project Work - {(int)NonProjectListBox.SelectedValue}";
+            }
+
             if (string.IsNullOrEmpty(projectID))
             {
                 MessageBox.Show("Project ID cannot be empty", "Error", MessageBoxButton.OK);
                 return;
             }
 
+            if ((NonProjectSelectionPanel.Visibility == Visibility.Visible && OtherSpecifyInput.Visibility == Visibility.Visible) && string.IsNullOrEmpty(OtherSpecifyInput.Text))
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_UI/CaptureProjectData.xaml.cs:182]: I would move this check to be above the building of the `projectID` variable above

File: BHoMAnalytics_UI/CaptureProjectData.xaml.cs [hunk 7]
```diff
@@ -103,10 +193,29 @@ private void ConfirmProject()
...
+            OtherSpecifyInput.Text = "";
+            OtherSpecifyInput.Visibility = Visibility.Collapsed;
+            OtherSpecifyText.Visibility = Visibility.Collapsed;
+
+            //Project
+            ProjectInputPanel.Visibility = Visibility.Hidden;
+            ProjectIDInput.Text = "";
+            ProjectBtn.Focus();
+
+            //Grasshopper specific
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_UI/CaptureProjectData.xaml.cs:213]: Is this still just Grasshopper now?

File: BHoMAnalytics_oM/Enums/NonProjectOption.cs [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        Development = 5,
+        Hackademy = 10,
+        Testing = 15,
+        Other = 20
+    }
+}
+
+
+
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_oM/Enums/NonProjectOption.cs:42]:
```suggestion
```

File: BHoMAnalytics_oM/Enums/NonProjectOption.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.oM.BHoMAnalytics
+{
+    public enum NonProjectOption
+    {
+        Development = 5,
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_oM/Enums/NonProjectOption.cs:34]: Tagging @alexissantella back in to confirm if they're happy with this, or would be better with integer increments and we add them in alphabetically but start at 1 and count up from there

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by FraserGreenroyd — APPROVED** [review]
Code review looks good, happy to merge if @alexissantella is happy with the enum.

File: BHoMAnalytics_oM/Enums/NonProjectOption.cs [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        Hackademy = 10,
+        Testing = 15,
+        Other = 20
```
Comment (FraserGreenroyd) [inline BHoMAnalytics_oM/Enums/NonProjectOption.cs:37]:
```suggestion
        Development = 1,
        Testing = 2,
        Training = 3,
        Hackademy = 4,
        Other = 5,
```

Discussed with @alexissantella on a call - this is the desired input for non-project workflows number wise, and adding the addition of `Training` as a discrete option to `Hackademy`.

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by travispotterBH** [comment]
@BHoMBot copyright-compliance

**Comment by travispotterBH** [comment]
@BHoMBot  check copyright-compliance

**Review by FraserGreenroyd — APPROVED** [review]
Code reviewed following call with Alexis happy to merge.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**